app/models/google_calendar/listing.rb

```diff
@@ -0,0 +1,81 @@
+module GoogleCalendar
+  module Listing
+    include Client
+
+    GOOGLE_EVENT_DECLINED_RESPONSE = 'declined'.freeze
+    LISTING_FIELDS = 'items(id, start, end, summary, recurrence, creator, attendees(self, responseStatus))'.freeze
+
+    def list_events(credentials, user_email, starting, ending)
+      all_events = daily_events_container
+      rooms = ConferenceRoom.all
+      calendar_service(credentials).batch do |service|
+        rooms.each do |room|
+          add_events_from_room(room, service, all_events, listing_options(starting, ending))
+        end
+      end
+      mark_user_events(user_email, all_events)
+      all_events
+    end
+
+    def daily_events_container
```

Comment:
I think you might benefit from creating a new class here instead of using a plain hash